addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,1024 @@
+                    getThing().getUID());
+            disconnect(socketChannel);
+        }
+
+        return commandList;
+    }
+
+    private String[] getRemoteList(int index) {
+
+        String getRemotes = "Agetremotes " + index + (char) 13;
+        ByteBuffer response = sendCommand(getRemotes);
+        String[] remoteList = new String[0];
+
+        if (response != null) {
+            String message = stripByteCount(response).split("\0")[0];
+            logger.trace("remotes returned {}", message);
+            if (message != null) {
+                if (message.contains("REMOTELIST")) {
+                    remoteList = message.split(",");
+                } else {
+                    logger.warn("Received some non-compliant command ({})", message);
+                    disconnect(socketChannel);
+                }
+            }
+        } else {
+            logger.warn("Did not receive an answer from the IRtrans transceiver for '{}' - Parsing is skipped",
+                    getThing().getUID());
+            disconnect(socketChannel);
+        }
+
+        return remoteList;
+    }
+
+    private ByteBuffer sendCommand(String command) {
+
+        if (command != null) {
+            ByteBuffer byteBuffer = ByteBuffer.allocate(command.getBytes().length);
+            try {
+                byteBuffer.put(command.getBytes("ASCII"));
+                onWritable(byteBuffer, socketChannel);
+                Thread.sleep(((BigDecimal) getConfig().get(RESPONSE_TIME_OUT)).intValue());
+                return onReadable(socketChannel, ((BigDecimal) getConfig().get(BUFFER_SIZE)).intValue());
+            } catch (UnsupportedEncodingException | NumberFormatException | InterruptedException e) {
+                logger.error("An exception occurred while configurting the IRtrans transceiver for '{}': {}",
+                        getThing().getUID(), e.getMessage());
+            }
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings("null")
+    private SocketChannel connect(String ipAddress, int portNumber) {
+        SocketChannel socketChannel = null;
+        try {
+            socketChannel = SocketChannel.open();
+        } catch (Exception e2) {
+            logger.error("An exception occurred while connecting to {}:{} : {}",
+                    new Object[] { ipAddress, portNumber, e2.getMessage() });
+        }
+
+        try {
+            socketChannel.socket().setKeepAlive(true);
+            socketChannel.configureBlocking(false);
+        } catch (Exception e) {
+            logger.error("An exception occurred while configuring the connection to '{}:{}' : {}",
+                    new Object[] { ipAddress, portNumber, e.getMessage() });
+        }
+
+        while (selector == null) {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                logger.error("An exception occurred while putting the Thread to sleep: {}", e.getMessage(), e);
```

Comment:
If we receive this InterruptedException, the interrupted flag was cleared and we most likely need to reset the flag:

http://www.yegor256.com/2015/10/20/interrupted-exception.html
https://www.ibm.com/developerworks/library/j-jtp05236/